### PR TITLE
optimized model for only calculating the extrinsic parameters

### PR DESCRIPTION
**Pull request by ahnitz**

### CODE REVIEW

**Comment by cdcapano** [comment]
@ahnitz Is this ready for review?

**Comment by ahnitz** [comment]
Not yet. I'd like to have some examples that show it works first.

**Comment by ahnitz** [comment]
@cdcapano I'm removing the WIP label on this one.

File: pycbc/inference/models/single_template.py [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+
+        if self.time is None:
+            self.time = p['tc']
+
+        shloglr = hhloglr = 0
+        for ifo in self.sh:
+            fp, fc = self.det[ifo].antenna_pattern(p['ra'], p['dec'],
+                                         p['polarization'], self.time)
+            dt = self.det[ifo].time_delay_from_earth_center(p['ra'],
+                                         p['dec'], self.time)
```
Comment (cdcapano) [inline pycbc/inference/models/single_template.py]: Fix PEP8 issues with indent.

File: pycbc/inference/models/single_template.py [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+            self.det[ifo] = Detector(ifo)
+            snr, _, _ = pyfilter.matched_filter_core(hp, data[ifo],
+                                            psd=psds[ifo],
+                                            low_frequency_cutoff=f_lower,
+                                            high_frequency_cutoff=f_upper)
+
+            self.sh[ifo] = 4 * df * snr
+            self.hh[ifo] = -0.5 * pyfilter.sigmasq(hp, psd=psds[ifo],
+                                              low_frequency_cutoff=f_lower,
+                                              high_frequency_cutoff=f_upper)
```
Comment (cdcapano) [inline pycbc/inference/models/single_template.py]: PEP8 indent issues.

File: pycbc/inference/models/single_template.py [hunk 3]
```diff
@@ -0,0 +1,108 @@
...
+from pycbc import filter as pyfilter
+from pycbc.waveform import get_fd_waveform
+from pycbc.detector import Detector
+
+from .base import BaseModel
+
+# In this model we only calculate terms up to a constant.
+# We are primarily interested in the posterior result
+
+class SingleTemplate(BaseModel):
```
[the remark this answers is not in the document]
Comment (ahnitz) [reply pycbc/inference/models/single_template.py]: BaseDataModel also requires a waveform generator which isn't really make sense here. The added methods also don't really make sense in this use case. The BaseModel seems to just do enough.

File: pycbc/inference/models/single_template.py [hunk 4]
```diff
@@ -0,0 +1,108 @@
...
+    maximizing over the extrinsic ones. We also assume a dominant mode waveform
+    approximant only and non-precessing.
+    """
+    name = 'single_template'
+
+    def __init__(self, data, psds, f_lower, f_upper=None, **kwargs):
+        # set up the boiler-plate attributes; note: we'll compute the
+        # log evidence later
+        super(SingleTemplate, self).__init__(**kwargs)
+        sample_rate = 32768
```
[the remark this answers is not in the document]
Comment (ahnitz) [reply pycbc/inference/models/single_template.py]: Yep, makes sense. Done

**Comment by ahnitz** [comment]
@cdcapano Can you check this again?

File: pycbc/inference/models/single_template.py (hunk 3 above)
[the remark this answers is not in the document]
Comment (cdcapano) [reply pycbc/inference/models/single_template.py]: Ok, I see your point about waveform generation. Still, it bothers me that you're calling log likelihood *ratio*, `loglikelihood`. The main point of `BaseData` is to introduce `loglr` and `lognl` for just this purpose. So... see #2468. That should address your problems. If so, please rebase once that's merged, and change the inheritance. I don't really care if you don't want to actually calculate `lognl`. Set it to 0 if you like; either way, at least someone who is only looking at the output files will be able to figure out what's going on.

**Comment by ahnitz** [comment]
@cdcapano Should be rebased now.

**Pull request merged**